src/uucore/src/lib/features/smack.rs

```diff
@@ -0,0 +1,74 @@
+// This file is part of the uutils uucore package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+// spell-checker:ignore smackfs
+//! SMACK (Simplified Mandatory Access Control Kernel) support
+
+use std::io;
+use std::path::Path;
+
+use thiserror::Error;
+
+use crate::error::{UError, strip_errno};
+use crate::translate;
+
+#[derive(Debug, Error)]
+pub enum SmackError {
+    #[error("{}", translate!("smack-error-not-enabled"))]
+    SmackNotEnabled,
+
+    #[error("{}", translate!("smack-error-label-retrieval-failure", "error" => strip_errno(.0)))]
+    LabelRetrievalFailure(io::Error),
+
+    #[error("{}", translate!("smack-error-label-set-failure", "context" => .0.clone(), "error" => strip_errno(.1)))]
+    LabelSetFailure(String, io::Error),
+}
+
+impl UError for SmackError {
+    fn code(&self) -> i32 {
+        match self {
+            Self::SmackNotEnabled => 1,
+            Self::LabelRetrievalFailure(_) => 2,
+            Self::LabelSetFailure(_, _) => 3,
+        }
+    }
+}
+
+impl From<SmackError> for i32 {
+    fn from(error: SmackError) -> Self {
+        error.code()
+    }
+}
+
+/// Checks if SMACK is enabled by verifying smackfs is mounted.
+pub fn is_smack_enabled() -> bool {
+    Path::new("/sys/fs/smackfs").exists()
```

Comment:
the result should be cached

Reply:
Want me to do the same for SELinux? I double checked it before adding a OnceCell, but can add this easily